### DISCUSSION TITLE
File finder: Only show hover state for mouse after movement vs. on panel open

### DISCUSSION THREAD

**Discussion by mkdynamic** [zed-industries/zed#36943] [Feature Requests]
### Description

If your mouse cursor is already over the area where the File finder panel opens, as soon as you open the panel that option is highlighted. If you are exclusively using the keyboard to type and select a file, this is quite distracting visually – since you have the actual selected item (via keyboard) and the hovered item (via mouse, which you never moved). So you end up with 2 "highlighted" items in the list.

Proposal would be to only activate the hovered state via mouse on an item after the mouse has moved. So a resting cursor would not cause the item to be highlighted.

**Comment by notpeter** [comment]
Hi @mkdynamic,

Thanks for filing this.
I understand this could be a distraction when two items are shown highlighted.
This is still an issue with the current Zed release.

This isn't something we're likely to prioritize development of at this time, but if someone is so inclined to fix this (don't initially highlight under the mouse cursor when Pickers are popped) I'd be willing to test and review PRs.

Thanks for filing.

**Comment by mgussekloo** [comment]
Hi, a bit late to the party but I had this same problem and wanted to share my workaround. I bind "cmd + p" (opening the File finder panel) to run a sequence of actions: first run a cli command that moves the mouse out of the way, then trigger the File finder panel. The task requires installing [cliclick](https://github.com/BlueM/cliclick) to move the mouse.

I'd still like a non-workaround solution, so consider this a +1.

```json
// Keymap

{ "bindings": {
      "cmd-p": ["action::Sequence", [
      	["task::Spawn", { "task_name": "mouse-move" }],
        ["file_finder::Toggle", { "separate_history": true }]
      ]],
}
```

```json
// Tasks

{
  	"label": "mouse-move",
   	"command": "cliclick",
    "args": ["m:-500,+0"],
    "reveal": "never",
    "hide": "always"
  }
```

**Comment by snowman** [comment]
If you are on Windows, here is the quick for autohotkey

```
#IfWinActive ahk_exe zed.exe

~^p::
  MouseMove, 0, %A_ScreenHeight%, 0
return

#IfWinActive
```

This solve the problem that when the mouse is in the `file finder` area, when press `Ctrl-p`, the mouse without movement auto select the pointed item, and then release `Ctrl-p` doesn't have a chance to typing to filter and that annoying random file is in front of you opened, make the file browsing history dirty
